src/scmdata/run.py

```diff
@@ -0,0 +1,1718 @@
+            elif col in self.meta_attributes:
+                keep_meta &= pattern_match(
+                    self._meta_column(col),
+                    values,
+                    regexp=regexp,
+                    has_nan=has_nan,
+                    separator=self.data_hierarchy_separator,
+                ).values
+            elif col == "year":
+                keep_ts &= years_match(self._time_points.years(), values)
+
+            elif col == "month":
+                keep_ts &= month_match(self._time_points.months(), values)
+
+            elif col == "day":
+                keep_ts &= self._day_match(values)
+
+            elif col == "hour":
+                keep_ts &= hour_match(self._time_points.hours(), values)
+
+            elif col == "time":
+                keep_ts &= datetime_match(self._time_points.values, values)
+
+            elif col == "level":
+                if "variable" not in filters.keys():
+                    keep_meta &= pattern_match(
+                        self._meta_column("variable"),
+                        "*",
+                        values,
+                        regexp=regexp,
+                        has_nan=has_nan,
+                        separator=self.data_hierarchy_separator,
+                    ).values
+                # else do nothing as level handled in variable filtering
+
+            else:
+                raise ValueError("filter by `{}` not supported".format(col))
+
+        return keep_ts, keep_meta
+
+    def _day_match(self, values):
+        if isinstance(values, str):
+            wday = True
+        elif isinstance(values, list) and isinstance(values[0], str):
+            wday = True
+        else:
+            wday = False
+
+        if wday:
+            days = self._time_points.weekdays()
+        else:  # ints or list of ints
+            days = self._time_points.days()
+
+        return day_match(days, values)
+
+    def head(self, *args, **kwargs):
+        """
+        Return head of :func:`self.timeseries()`.
+
+        Parameters
+        ----------
+        *args
+            Passed to :func:`self.timeseries().head()`
+
+        **kwargs
+            Passed to :func:`self.timeseries().head()`
+
+        Returns
+        -------
+        :obj:`pd.DataFrame`
+            Tail of :func:`self.timeseries()`
+        """
+        return self.timeseries().head(*args, **kwargs)
+
+    def tail(self, *args: Any, **kwargs: Any) -> pd.DataFrame:
+        """
+        Return tail of :func:`self.timeseries()`.
+
+        Parameters
+        ----------
+        *args
+            Passed to :func:`self.timeseries().tail()`
+
+        **kwargs
+            Passed to :func:`self.timeseries().tail()`
+
+        Returns
+        -------
+        :obj:`pd.DataFrame`
+            Tail of :func:`self.timeseries()`
+        """
+        return self.timeseries().tail(*args, **kwargs)
+
+    def rename(self, mapping: Dict[str, Dict[str, str]], inplace: bool = False):
```

Comment:
deprecate this?

Reply:
I'll straight up delete it. The docs are a lie anyway